### PR TITLE
Simplify block checks

### PR DESCRIPTION
**Pull request by Camwyn**
This restructures our work around Editor.php to make the checks simpler, more "canonical", and easier to expand for compatibilities.

Includes compatibility with the Classic Editor plugin.

Required by:
(list here as I open the PRs)

[TEC-4016]

[TEC-4016]: https://theeventscalendar.atlassian.net/browse/TEC-4016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+	 *
+	 * @var array<string>
+	 */
+	public static $classic_values  = [
+		'replace',
+		'classic',
+	];
+
+	/**
+	 * Placeholders
```
Comment (jesseeproductions) [inline src/Tribe/Editor/Compatibility/Classic_Editor.php:101]: Does this comment have to be updated for the properties?

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+	/**
+	 * Hooks for loading logic outside this class.
+	 *
+	 * @since TBD
+	 */
+	public function hooks() {
+		add_action( 'tribe_plugins_loaded', [ $this, 'set_classic_url_params' ], 22 );
+
+		global $current_screen;
+		$good_screens = [
```
Comment (jesseeproductions) [inline src/Tribe/Editor/Compatibility/Classic_Editor.php:132]: Should we add a check here if the current screen is empty?

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+		if ( self::get_classic_override() ) {
+			$should_load_blocks = true;
+		}
+
+		if ( self::get_classic_param() ) {
+			$should_load_blocks = false;
+		}
+
+		global $pagenow;
+
```
Comment (jesseeproductions) [inline src/Tribe/Editor/Compatibility/Classic_Editor.php:226]: How about another check here that the global is an array or object?

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+		$id = isset(  $_GET[ 'post' ] ) ? (int) $_GET[ 'post' ] : null;
+
+		$remember = get_post_meta( $id, static::$post_meta_key, true );
+
+		if ( empty( $remember ) ) {
+			return false;
+		}
+
+		// Why WP, why did you use a different term here?
+		return str_replace( '-editor', '', $remember );
```
Comment (jesseeproductions) [inline src/Tribe/Editor/Compatibility/Classic_Editor.php:358]: To make life difficult!

File: src/Tribe/Main.php [hunk 5]
```diff
@@ -657,6 +657,7 @@ public function bind_implementations() {
 		tribe_register_provider( Tribe\Service_Providers\Onboarding::class );
 		tribe_register_provider( Tribe\Admin\Notice\Service_Provider::class );
 		tribe_register_provider( Tribe\Admin\Conditional_Content\Service_Provider::class );
+		$bar = '';
```
Comment (jesseeproductions) [inline src/Tribe/Main.php:660]: Dev coding?

**Comment by jesseeproductions** [comment]
Nice! I look forward to better support for the classic editor!

File: src/Tribe/Settings_Manager.php [hunk 6]
```diff
@@ -163,6 +163,20 @@ public static function set_option( $name, $value ) {
...
+	 *
+	 * @param string $name
+	 *
+	 * @return bool
+	 */
+	public static function remove_option( $name ) {
+		$options          = self::get_options();
+		unset( $options[ $name ] );
+
+		return self::set_options( $options );
```
Comment (bordoni) [inline src/Tribe/Settings_Manager.php:177]: Use static?

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+	 * @since TBD
+	 *
+	 * @return bool|string The string of the editor choice or false on fails.
+	 */
+	public static function classic_editor_remembers() {
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		$id = isset(  $_GET[ 'post' ] ) ? (int) $_GET[ 'post' ] : null;
```
Comment (bordoni) [inline src/Tribe/Editor/Compatibility/Classic_Editor.php:349]: `tribe_get_request_var`?

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+
+		if ( false !== $remember ) {
+			$should_load_blocks = static::$block_term === $remember;
+		}
+
+		if ( self::get_classic_override() ) {
+			$should_load_blocks = true;
+		}
+
+		if ( self::get_classic_param() ) {
```
Comment (bordoni) [inline src/Tribe/Editor/Compatibility/Classic_Editor.php:221]: static.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor/Compatibility/Classic_Editor.php (hunk 2 above)
Comment (Camwyn) [reply src/Tribe/Editor/Compatibility/Classic_Editor.php:132]: Actually, this needs to be removed, this function gets called a few times before `$current_screen` is available 🤦🏼

File: src/Tribe/Editor/Compatibility/Classic_Editor.php (hunk 7 above)
Comment (Camwyn) [reply src/Tribe/Editor/Compatibility/Classic_Editor.php:349]: not always available

File: src/Tribe/Settings_Manager.php (hunk 6 above)
Comment (Camwyn) [reply src/Tribe/Settings_Manager.php:177]: I used self because of set option above, but yes.

**Review by bordoni — APPROVED** [review]

**Review by bordoni — APPROVED** [review]

**Pull request merged**